runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkBatchTransformTranslators.java

```diff
@@ -413,12 +411,15 @@ public void translateNode(
               outputType,
               FlinkIdentityFunction.of(),
               getCurrentTransformName(context));
-      final Configuration partitionOptions = new Configuration();
-      partitionOptions.setString(
-          Optimizer.HINT_SHIP_STRATEGY, Optimizer.HINT_SHIP_STRATEGY_REPARTITION);
```

Comment:
Commented on the bug, but _if_ Flink really looks at these hints, I think the implementation here is good. It is better to translate it as a hint that Flink understands than to force it.

Reply:
Ah I see from https://nightlies.apache.org/flink/flink-docs-release-1.14/docs/dev/dataset/transformations/#rebalance that it may just be even redistribution, which would be ignoring the hint really.

Reply:
Yes, I saw the problem appearing on a real pipeline. SDF uses Reshuffle to distribute restrictions among workers. My pipelime ended processing all restrictions on single worker, effectively detroying any parallelism.